zcash_client_sqlite/src/lib.rs

```diff
@@ -1238,35 +1289,27 @@ impl<P: consensus::Parameters> WalletWrite for WalletDb<rusqlite::Connection, P>
                 wallet::transparent::mark_transparent_utxo_spent(wdb.conn.0, tx_ref, &txin.prevout)?;
             }
 
-            // If we have some transparent outputs:
-            if d_tx
-                .tx()
-                .transparent_bundle()
-                .iter()
-                .any(|b| !b.vout.is_empty())
```

Comment:
Notice that we were checking whether there are any elements to loop over before looping over them, which is pointless. It would not be pointless if we were avoiding running the code below that handles funding accounts. However, that code is a duplicate of [lines 1101 to 1111](https://github.com/zcash/librustzcash/pull/1257/files#diff-57d91750cf8573e5cfebdee8bd31b2d635cb02adf7f5e5fdd20ff2253e43782fR1101-R1111) and so can also be deleted from here (nothing intervening between there and here can affect its results, and it is unnecessary to log the warning more than once).